scripts/mac/functions.bash

```diff
@@ -0,0 +1,274 @@
+    echo -e " --help                       Show usage"
+}
+
+# Gets arguments
+# Input:
+#   Agent script arguments ($@)
+# Output:
+#   app_url - Logz.io app URL
+#   agent_id - Logz.io agent ID
+#   app_json_file - App JSON (only in debug)
+# Error:
+#   Exit Code 2
+function get_arguments () {
+    write_log "INFO" "Getting arguments ..."
+
+    while true; do
+        case "$1" in
+            --help)
+                show_help
+                delete_temp_dir
+                exit
+                ;;
+            --url=*)
+                app_url=$(echo "$1" | cut -d "=" -f2)
+                if [[ "$app_url" = "" ]]; then
+                    print_error "agent.bash (2): no Logz.io app URL specified!"
+                    delete_temp_dir
+                    exit 2
+                fi
+
+                write_log "INFO" "url = $app_url"
+                ;;
+            --id=*)
+                agent_id=$(echo "$1" | cut -d "=" -f2)
+                if [[ "$agent_id" = "" ]]; then
+                    print_error "agent.bash (2): no agent ID specified!"
+                    delete_temp_dir
+                    exit 2
+                fi
+
+                write_log "INFO" "id = $agent_id"
+                ;;
+            --debug=*)
+                app_json_file=$(echo "$1" | cut -d "=" -f2)
+                if [[ "$app_json_file" = "" ]]; then
+                    print_error "agent.bash (2): no JSON file specified!"
+                    delete_temp_dir
+                    exit 2
+                fi
+
+                write_log "INFO" "debug = $app_json_file"
+                break
+                ;;
+            "")
+                break
+                ;;
+            *)
+                print_error "agent.bash (2): unrecognized flag"
+                print_error "agent.bash (2): try './agent.bash --help' for more information"
+                delete_temp_dir
+                exit 2
+                ;;
+        esac
+        shift
+    done
+
+    check_validation
+}
+
+# Checks validation of the arguments
+# Error:
+#   Exit Code 2
+function check_validation () {
+    write_log "INFO" "Checking validation ..."
+
+    if [[ ! -z "$app_json_file" ]]; then
+        if [[ -f "$app_json_file" ]]; then
+            return
+        fi
+
+        print_error "agent.bash (2): the JSON file $app_json_file does not exist"
+        delete_temp_dir
+        exit 2
+    fi
+
+    local is_error=false
+
+    if [[ -z "$app_url" ]]; then
+        is_error=true
+        print_error "agent.bash (2): Logz.io app URL must be specified"
+    fi
+    if [[ -z "$agent_id" ]]; then
+        is_error=true
+        print_error "agent.bash (2): agent ID must be specified"
+    fi
+
+    if $is_error; then
+        print_error "agent.bash (2): try './agent.bash --help' for more information"
+        delete_temp_dir
+        exit 2
+    fi
+}
+
+# Installs Homebrew
+# Error:
+#   Exit Code 3
+function install_homebrew () {
```

Comment:
why we need homebrew?

Reply:
if there is no jq installed and no homebrew installed, we will install homebrew to install jq.